### ISSUE TITLE
Upgrade 0.30.4 to 0.31.1 / query builder queries fail

### ISSUE THREAD

**Issue by mdz0r** [metabase/metabase#9017]
Hello,

Upgraded this morning to 0.31.1. In dashboards with filters, all my queries built with the query builder fail with :
error "Unknown column 'field' in 'where clause'",

SQL written queries still work.

```
nov. 26 15:57:38 WARN metabase.query-processor :: {:status :failed,
 :class com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException,
 :error "Unknown column 'field' in 'where clause'",
 :stacktrace
 ("sun.reflect.GeneratedConstructorAccessor85.newInstance(Unknown Source)"
  "sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)"
  "java.lang.reflect.Constructor.newInstance(Constructor.java:423)"
  "com.mysql.jdbc.Util.handleNewInstance(Util.java:425)"
  "com.mysql.jdbc.Util.getInstance(Util.java:408)"
  "com.mysql.jdbc.SQLError.createSQLException(SQLError.java:944)"
  "com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)"
  "com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)"
  "com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)"
  "com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)"
  "com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2484)"
  "com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1858)"
  "com.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:1966)"
  "com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:353)"
  "clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1002)"
  "clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:996)"
  "clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1025)"
  "clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1005)"
  "clojure.java.jdbc$query.invokeStatic(jdbc.clj:1099)"
  "clojure.java.jdbc$query.invoke(jdbc.clj:1056)"
  "toucan.db$query.invokeStatic(db.clj:275)"
  "toucan.db$query.doInvoke(db.clj:271)"
  "clojure.lang.RestFn.invoke(RestFn.java:410)"
  "toucan.db$simple_select.invokeStatic(db.clj:379)"
  "toucan.db$simple_select.invoke(db.clj:368)"
  "toucan.db$simple_select_one.invokeStatic(db.clj:405)"
  "toucan.db$simple_select_one.invoke(db.clj:394)"
  "toucan.db$select_one.invokeStatic(db.clj:606)"
  "toucan.db$select_one.doInvoke(db.clj:599)"
  "clojure.lang.RestFn.applyTo(RestFn.java:139)"
  "clojure.core$apply.invokeStatic(core.clj:659)"
  "clojure.core$apply.invoke(core.clj:652)"
  "toucan.db$select_one_field.invokeStatic(db.clj:615)"
  "toucan.db$select_one_field.doInvoke(db.clj:608)"
  "clojure.lang.RestFn.invoke(RestFn.java:464)"
  "--> query_processor.middleware.parameters.mbql$parse_param_value_for_type.invokeStatic(mbql.clj:22)"
  "query_processor.middleware.parameters.mbql$parse_param_value_for_type.invoke(mbql.clj:13)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856$fn__32866.invoke(mbql.clj:56)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856.invoke(mbql.clj:38)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856$fn__32866$iter__32871__32875$fn__32876$fn__32877.invoke(mbql.clj:45)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856$fn__32866$iter__32871__32875$fn__32876.invoke(mbql.clj:44)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856$fn__32866.invoke(mbql.clj:44)"
  "query_processor.middleware.parameters.mbql$fn__32851$build_filter_clause__32856.invoke(mbql.clj:38)"
  "query_processor.middleware.parameters.mbql$expand.invokeStatic(mbql.clj:71)"
  "query_processor.middleware.parameters.mbql$expand.invoke(mbql.clj:58)"
  "query_processor.middleware.parameters$expand_parameters_STAR_.invokeStatic(parameters.clj:18)"
  "query_processor.middleware.parameters$expand_parameters_STAR_.invoke(parameters.clj:12)"
  "query_processor.middleware.parameters$expand_parameters.invokeStatic(parameters.clj:43)"
  "query_processor.middleware.parameters$expand_parameters.invoke(parameters.clj:40)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invokeStatic(parameters.clj:49)"
  "query_processor.middleware.parameters$substitute_parameters_STAR_.invoke(parameters.clj:46)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__32156.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__35279.invoke(resolve_driver.clj:15)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31351$fn__31352.invoke(bind_effective_timezone.clj:9)"
  "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
  "util.date$call_with_effective_timezone.invoke(date.clj:77)"
  "query_processor.middleware.bind_effective_timezone$bind_effective_timezone$fn__31351.invoke(bind_effective_timezone.clj:8)"
  "query_processor.middleware.store$initialize_store$fn__37496$fn__37497.invoke(store.clj:11)"
  "query_processor.store$do_with_new_store.invokeStatic(store.clj:34)"
  "query_processor.store$do_with_new_store.invoke(store.clj:30)"
  "query_processor.middleware.store$initialize_store$fn__37496.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__31773.invoke(cache.clj:127)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__31821.invoke(catch_exceptions.clj:64)"
  "query_processor$process_query.invokeStatic(query_processor.clj:213)"
  "query_processor$process_query.invoke(query_processor.clj:209)"
  "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:322)"
  "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
  "query_processor$fn__37661$process_query_and_save_execution_BANG___37666$fn__37667.invoke(query_processor.clj:354)"
  "query_processor$fn__37661$process_query_and_save_execution_BANG___37666.invoke(query_processor.clj:340)"
  "api.card$run_query_for_card.invokeStatic(card.clj:580)"
  "api.card$run_query_for_card.doInvoke(card.clj:566)"
  "api.card$fn__46963$fn__46966.invoke(card.clj:587)"
  "api.card$fn__46963.invokeStatic(card.clj:586)"
  "api.card$fn__46963.invoke(card.clj:582)"
  "middleware$enforce_authentication$fn__56091.invoke(middleware.clj:113)"
  "api.routes$fn__56237.invokeStatic(routes.clj:62)"
  "api.routes$fn__56237.invoke(routes.clj:62)"
  "routes$fn__56326$fn__56327.doInvoke(routes.clj:108)"
  "routes$fn__56326.invokeStatic(routes.clj:103)"
  "routes$fn__56326.invoke(routes.clj:103)"
  "middleware$catch_api_exceptions$fn__56226.invoke(middleware.clj:436)"
  "middleware$log_api_call$fn__56204$fn__56206.invoke(middleware.clj:364)"
  "middleware$log_api_call$fn__56204.invoke(middleware.clj:363)"
  "middleware$add_security_headers$fn__56146.invoke(middleware.clj:252)"
  "core$wrap_streamed_json_response$fn__62590.invoke(core.clj:67)"
  "middleware$bind_current_user$fn__56096.invoke(middleware.clj:137)"
  "middleware$maybe_set_site_url$fn__56156.invoke(middleware.clj:290)"
  "middleware$add_content_type$fn__56149.invoke(middleware.clj:262)"),
 :query
 {:query {:source-table 5, :aggregation [[:metric 2]], :breakout [[:fk-> [:field-id 46] [:field-id 61]]]},
  :type :query,
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :parameters [{:type "date/all-options", :target ["dimension" ["field-id" 42]], :value "2018-01-01~2018-07-01"} {:type "id", :target ["dimension" ["fk->" 46 60]], :value ["006"]}],
  :middleware nil,
  :cache-ttl nil,
  :info
  {:executed-by 1,
   :context :question,
   :card-id 14,
   :dashboard-id nil,
   :query-hash [115, -51, 84, -53, 94, 76, 3, 70, -111, -27, 94, 51, 112, -50, 115, 46, -48, 21, 93, -18, 74, -53, -64, 121, 59, -59, 26, 96, -113, -60, 73, 51],
   :query-type "MBQL"}},
 :preprocessed nil,
 :native nil}

nov. 26 15:57:38 WARN metabase.query-processor :: Query failure: Unknown column 'field' in 'where clause'
("clojure.core$ex_info.invokeStatic(core.clj:4739)"
 "clojure.core$ex_info.invoke(core.clj:4739)"
 "--> query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:285)"
 "query_processor$assert_query_status_successful.invoke(query_processor.clj:277)"
 "query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:323)"
 "query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:316)"
 "query_processor$fn__37661$process_query_and_save_execution_BANG___37666$fn__37667.invoke(query_processor.clj:354)"
 "query_processor$fn__37661$process_query_and_save_execution_BANG___37666.invoke(query_processor.clj:340)"
 "api.card$run_query_for_card.invokeStatic(card.clj:580)"
 "api.card$run_query_for_card.doInvoke(card.clj:566)"
 "api.card$fn__46963$fn__46966.invoke(card.clj:587)"
 "api.card$fn__46963.invokeStatic(card.clj:586)"
 "api.card$fn__46963.invoke(card.clj:582)"
 "middleware$enforce_authentication$fn__56091.invoke(middleware.clj:113)"
 "api.routes$fn__56237.invokeStatic(routes.clj:62)"
 "api.routes$fn__56237.invoke(routes.clj:62)"
 "routes$fn__56326$fn__56327.doInvoke(routes.clj:108)"
 "routes$fn__56326.invokeStatic(routes.clj:103)"
 "routes$fn__56326.invoke(routes.clj:103)"
 "middleware$catch_api_exceptions$fn__56226.invoke(middleware.clj:436)"
 "middleware$log_api_call$fn__56204$fn__56206.invoke(middleware.clj:364)"
 "middleware$log_api_call$fn__56204.invoke(middleware.clj:363)"
 "middleware$add_security_headers$fn__56146.invoke(middleware.clj:252)"
 "core$wrap_streamed_json_response$fn__62590.invoke(core.clj:67)"
 "middleware$bind_current_user$fn__56096.invoke(middleware.clj:137)"
 "middleware$maybe_set_site_url$fn__56156.invoke(middleware.clj:290)"
 "middleware$add_content_type$fn__56149.invoke(middleware.clj:262)")

```

**Comment by ans-4175** [comment]
Is it same as https://github.com/metabase/metabase/issues/8932 ?

**Comment by mdz0r** [comment]
Doesnt seem to.

It's not only related to date fields.

**Comment by camsaul** [comment]
Fixed by #9025
